ui/src/adapters/api/credentials.ts

```diff
@@ -52,30 +52,92 @@ const proofTypeParser = getStrictParser<ProofTypeInput[], ProofType[]>()(
-  createdAt: string;
-  expiresAt: string | null;
-  proofTypes: ProofTypeInput[];
+type CredentialInput = Pick<Credential, "id" | "revoked" | "schemaHash"> & {
+  proofTypes: CredentialProofType[];
+  vc: {
+    credentialSchema: {
+      id: string;
+    } & Record<string, unknown>;
+    credentialStatus: {
+      revocationNonce: number;
+    } & Record<string, unknown>;
+    credentialSubject: {
+      type: string;
+    } & Record<string, unknown>;
+    expirationDate?: string | null;
+    issuanceDate: string;
+    issuer: string;
+    refreshService?: RefreshService | null;
+  };
 };
 
 export const credentialParser = getStrictParser<CredentialInput, Credential>()(
-  z.object({
-    createdAt: datetimeParser,
-    credentialSubject: z.record(z.unknown()),
-    expired: z.boolean(),
-    expiresAt: datetimeParser.nullable(),
-    id: z.string(),
-    proofTypes: proofTypeParser,
-    refreshService: z
-      .object({ id: z.string(), type: z.literal("Iden3RefreshService2023") })
-      .nullable(),
-    revNonce: z.number(),
-    revoked: z.boolean(),
-    schemaHash: z.string(),
-    schemaType: z.string(),
-    schemaUrl: z.string(),
-    userID: z.string(),
-  })
+  z
+    .object({
+      id: z.string(),
+      proofTypes: proofTypeParser,
+      revoked: z.boolean(),
+      schemaHash: z.string(),
+      vc: z.object({
+        credentialSchema: z
+          .object({
+            id: z.string(),
+          })
+          .and(z.record(z.unknown())),
+        credentialStatus: z
+          .object({
+            revocationNonce: z.number(),
+          })
+          .and(z.record(z.unknown())),
+        credentialSubject: z
+          .object({
+            type: z.string(),
+          })
+          .and(z.record(z.unknown())),
+        expirationDate: datetimeParser.nullable().default(null),
+        issuanceDate: datetimeParser,
+        issuer: z.string(),
+        refreshService: z
+          .object({ id: z.string(), type: z.literal("Iden3RefreshService2023") })
+          .nullable()
+          .default(null),
+      }),
+    })
+    .transform(
+      ({
+        id,
+        proofTypes,
+        revoked,
+        schemaHash,
+        vc: {
+          credentialSchema,
+          credentialStatus,
+          credentialSubject,
+          expirationDate,
+          issuanceDate,
+          issuer,
+          refreshService,
+        },
+      }) => {
+        const expired = expirationDate ? new Date() > new Date(expirationDate) : false;
+
+        return {
+          createdAt: issuanceDate,
+          credentialSubject,
+          expired,
+          expiresAt: expirationDate,
+          id,
+          proofTypes,
+          refreshService: refreshService,
+          revNonce: credentialStatus.revocationNonce,
+          revoked,
+          schemaHash,
+          schemaType: credentialSubject.type,
+          schemaUrl: credentialSchema.id,
+          userID: issuer,
+        };
```

Comment:
Now that the API has changed that drastically, maybe we can consider updating our domain type, so our structure better matches what the API provides.

For example, mapping `createdAt` to `issuanceDate` does not make much sense, since the previous value was a database record creation date while the current value represents when the credential was issued.

I would create a parser for the vc object that processes the types that require it, like dates, and use it on the wrapper type that would better match the current API structure.

Reply:
Maybe I did not explain myself well. I would create a separate type and parser for the `vc` object and then add a `vc` property to our domain Credential type, so we don't need to do all this flattening.

Reply:
can we make this refactoring as a tech debt in the future PR to finish this one?

Reply:
Ok, please ping me when the PR is ready. I'll leave this comment open here.